### ISSUE TITLE
OCaml example from readme

### ISSUE THREAD

**Issue by gtach2o** [shawn-mcginty/naboris#31]
Hi!
I was trying to execute OCaml example from readme
```OCaml
(* OCaml *)
let server_config: unit Naboris.ServerConfig.t = Naboris.ServerConfig.create ()
  |> Naboris.ServerConfig.setRequestHandler(fun route req res ->
    match (Naboris.Route.path route) with
      | ["hello"] ->
        res
          |> Naboris.Res.text req "Hello world!";
      | _ ->
        res
          |> Naboris.Res.status 404
          |> Naboris.Res.text req "Resource not found.";
  ) in

Naboris.listenAndWaitForever 3000 server_config
(* In a browser navigate to http://localhost:3000/hello *)
```
but I didn't get success. 

Here are the steps:

```bash
opam switch create . ocaml-base-compiler.4.09.0
eval $(opam env)
opam install merlin ocp-indent dune utop naboris
touch main.ml #paste OCaml example
touch dune #paste dune config
```
```
(executable
    (name main)
    (libraries naboris)
)
```
```bash
dune build main.exe
```
No errors.
Execute 
```bash
dune exec ./main.exe 
# or
_build/default/main.exe 
```
nothing happens. 
I'm beginner in OCaml, so maybe I did something wrong. I would appreciate any advice. Thanks.

**Comment by shawn-mcginty** [comment]
@gtach2o thanks for checking out naboris!

I will definitely update the readme to an actual working example.. that's my fault.

`Naboris.listenAndWaitForever` actually returns an `Lwt` `Promise` (which never resolves)

At the highest level `Lwt` needs to be executed within the context of `Lwt_main` . You can read their documentation here https://ocsigen.org/lwt/4.1.0/api/Lwt_main 

To get your example up and running wrap the last call like this.
```
let _ = Lwt_main.run begin
  Naboris.listenAndWaitForever 3000 server_config
end
```

**Comment by gtach2o** [comment]
Thank you! It works.

**Comment by shawn-mcginty** [comment]
I'm going to leave this open until I actually fix the README

**Comment by shawn-mcginty** [comment]
Will be fixed in `0.1.1`
